From 4391713d64dd8f8b5033a0ad0338144cb0571e84 Mon Sep 17 00:00:00 2001
From: Andrey Borzenkov <arvidjaar@gmail.com>
Date: Thu, 23 Feb 2012 16:10:33 +0000
Subject: [PATCH 501/509] apply TimeoutSec to oneshot services too

Make sure timeout is applied also to oneshot services. We do have
examples of long running processes that wait for external conditions
(plymouth-quit-wait as example).

Signed-off-by: Andrey Borzenkov <arvidjaar@gmail.com>
---
 src/service.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/service.c b/src/service.c
index bf2e0a9..1dcd993 100644
--- a/src/service.c
+++ b/src/service.c
@@ -2110,7 +2110,7 @@ static void service_enter_start(Service *s) {
 
         if ((r = service_spawn(s,
                                c,
-                               s->type == SERVICE_FORKING || s->type == SERVICE_DBUS || s->type == SERVICE_NOTIFY,
+                               s->type == SERVICE_FORKING || s->type == SERVICE_DBUS || s->type == SERVICE_NOTIFY || s->type == SERVICE_ONESHOT,
                                true,
                                true,
                                true,
-- 
1.7.9.3

